core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -165,10 +176,41 @@ public List<RawNotarizedTransaction> getTransactionsForProposal(
                         .map(RawNotarizedTransaction::create))
             .toList();
 
+    final var rawPreviousExecutedTransactionsTotalSize =
+        rawPreviousExecutedTransactions.stream()
+            .map(tx -> tx.getPayload().length)
+            .reduce(0, Integer::sum);
+    final var remainingSizeInUncommittedVertices =
+        maxProposalAndUncommittedVerticesTotalTxnPayloadSize
+            - rawPreviousExecutedTransactionsTotalSize;
+
+    final var maxPayloadSize =
+        Math.min(remainingSizeInUncommittedVertices, maxProposalTotalTxnsPayloadSize);
+
+    metrics.ledger().maxProposalPayloadSize().inc(maxPayloadSize);
+
     // TODO: Don't include transactions if NextEpoch is to occur
     // TODO: This will require Proposer to simulate a NextRound update before proposing
-    return stateComputer.getTransactionsForProposal(
-        maxNumTransactionsPerProposal, maxProposalTotalTxnsPayloadSize, transactionsToExclude);
+    final var result =
+        maxPayloadSize > 0 && maxNumTransactionsPerProposal > 0
+            ? stateComputer.getTransactionsForProposal(
+                maxNumTransactionsPerProposal, maxPayloadSize, rawPreviousExecutedTransactions)
+            : List.<RawNotarizedTransaction>of();
+
+    final var resultTotalTxnPayloadSize =
+        result.stream().map(tx -> tx.getPayload().length).reduce(0, Integer::sum);
+
+    final var totalUncommittedTxnBytesIncludingThisProposal =
+        resultTotalTxnPayloadSize + rawPreviousExecutedTransactionsTotalSize;
+
+    metrics.ledger().numTransactionsIncludedInProposal().inc(result.size());
```

Comment:
These metrics should really be Summaries (or histograms - although we don't have support for these in Java), not Counters - summaries track both "total" and "count" - see: https://prometheus.io/docs/practices/histograms/

Reply:
Out of interest why did you put these in `ledger`, rather than `bft`?

It's also a little weird they are leader-only metrics? Shouldn't these metrics be incremented during proposal processing, so they can be recorded by all nodes in consensus?

If you want to keep them here, they should include `leader` in their name imo 👍

Reply:
Yeah my idea was to report these values for each node as a leader.
:+1: for Summaries
:+1: for moving to `bft`
:+1: for prepending `leader`

Reply:
Sounds good - in a future ticket, I think we should _also_ report these values as someone receiving the proposals - to ensure we can EG debug mainnet using our foundation validators or understand if validators are acting up :)